File: lib/kwartz/binding/eruby.rb
###
### $Rev$
### $Release$
### $Copyright$
###

require 'kwartz/converter'
require 'kwartz/translator'
require 'kwartz/binding/ruby'



module Kwartz



  ##
  ## directive handler for eRuby
  ##
  class ErubyHandler < RubyHandler
  end
  Handler.register_class('eruby', ErubyHandler)



  ##
  ## translator for eRuby
  ##
  class ErubyTranslator < BaseTranslator


    ERUBY_EMBED_PATTERNS = [
      '<% ',    ' %>',        # statement
      '<%= ',   ' %>',        # expression
      '<%=h ',  ' %>',        # escaped expression
    ]


    def initialize(properties={})
      super(ERUBY_EMBED_PATTERNS, properties)
    end


  end #class
  Translator.register_class('eruby', ErubyTranslator)



end #module

File: Kwartz/Binding/Eruby.php
<?php
// vim: set expandtab tabstop=4 shiftwidth=4 softtabstop=4:

// $Rev$
// $Release$
// $Copyright$

require_once 'Kwartz/Converter.php';
require_once 'Kwartz/Translator.php';
require_once 'Kwartz/Binding/Ruby.php';


/**
 *  directive handler for eRuby
 */
class KwartzErubyHandler extends KwartzRubyHandler {
}



/**
 *  translator for eRuby
 */
class KwartzErubyTranslator extends KwartzBaseTranslator {


    function __construct($properties=array()) {
        $marks = array(
            '<% ',   ' %>',    // statement
            '<%= ',  ' %>',    // expression
            '<%=h ', ' %>',    // escaped expression
            );
        parent::__construct($marks, $properties);
    }


}


?>
